### PR TITLE
#669-Bug: Analog time doesn't correspond with digital time

### LINKED ISSUE #669: Bug: Analog time doesn't correspond with digital time
### Describe the bug.

The analog time doesn't correspond with digital time in some countries. E..g America /Anchorage 

### Steps to reproduce.
* Launch the app.

* Click on the menu and go to utilities 

* Click on word clock  

*  Then select  country to "America/Anchorage"

* And note the bug

### Expected behavior 

I expected that the analog and digital time should be the same. 

### Screen recording
https://youtu.be/BaY4AQQBw7s

### Devices and Versions):)
* Device: [Gionee]

* Version [7.0]

* App version [5.5.0]

### PR DESCRIPTION
**Pull request by HeinKhantZaw**
## Description

Issue #669 is fixed. I fixed the wrong time error for both analog clock and digital clock in world clock utility. Now, the clock works correctly whatever the device's local time is (for both 12 hour format and 24 hour format).

**Required Dependency**
com.tomerrosenfeld.customanalogclockview:custom-analog-clock-view:1.1

Fixes #669 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/VerifyEmailActivity.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void verifyEmail(String code, boolean verify) {
         Log.v("EXECUTING", uri);
         progressDialog.show();
 
-        //Set up client
+        //Set up Client
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/VerifyEmailActivity.java]: you should do the changes applicable only for this PR only.

No need of just capitalizing a letter in file

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/description/RestaurantsActivity.java [hunk 2]
```diff
@@ -196,7 +196,7 @@ private void getRestaurantItems() {
         String uri = API_LINK_V2 + "get-all-restaurants/" + mCity.getLatitude() + "/" + mCity.getLongitude();
         Log.v(TAG, "URI : " + uri);
 
-        //Set up client
+        //Set up Client
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/description/RestaurantsActivity.java]: please revert the files which only have this chnage

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyConverterAdapter.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
                 CurrencyConverterGlobal.global_country_name = mListCurrencyNamesFiltered.get(position).shortName;
                 CurrencyConverterGlobal.country_id = mListCurrencyNamesFiltered.get(position).abbreviation;
 
-            } else { // Is This is for world clock
+            } else { // Is This is for world Clock
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyConverterAdapter.java]: revert this

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java [hunk 4]
```diff
@@ -2,44 +2,38 @@
...
-import android.util.Log;
 import android.view.MenuItem;
-import android.widget.ImageView;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
 import android.widget.TextClock;
-import android.widget.TextView;
+import com.tomerrosenfeld.customanalogclockview.CustomAnalogClock;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: What was the issue with earlier code?
We won't be adding a new dependency for world clock.

File: Android/app/src/main/java/io/github/project_travel_mate/VerifyEmailActivity.java (hunk 1 above)
Comment (HeinKhantZaw) [reply Android/app/src/main/java/io/github/project_travel_mate/VerifyEmailActivity.java]: Sry, I didn't notice this change, all I did is editing the code in WorldClockActivity.java.
I will take a look back.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java (hunk 4 above)
Comment (HeinKhantZaw) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: In, the earlier code, the original library for the analog clock has issues (shows glitch effects and incorrect time) when the users choose another time zone. And also, the previous contributor for the world clock utility set the device local time as Kolkata's time. This becomes error when Users from different time zone use this utility.
So, I replaced with another analog library for showing the correct time.

**Comment by HeinKhantZaw** [comment]
I am really sorry about this. I will edit the files back. I didn't actually notice that the comments have changed.

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyConverterAdapter.java (hunk 3 above)
Comment (HeinKhantZaw) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/CurrencyConverterAdapter.java]: I have changed it, you can see in commits.

**Comment by HeinKhantZaw** [comment]
I replaced drawable files with png files because the analog clock library only accepts png.

**Comment by Swati4star** [comment]
@HeinKhantZaw 
- Please dont add a new library to fix this issue
- resolve merge conflicts
- keep only relevant chnages in this file

File: Android/app/src/main/java/io/github/project_travel_mate/friend/FriendsProfileActivity.java [hunk 5]
```diff
@@ -349,4 +349,4 @@ private void networkError() {
         animationView.playAnimation();
     }
 
-}
\ No newline at end of file
+}
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/friend/FriendsProfileActivity.java]: these files shouldn't be a part of this PR

File: Android/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -272,7 +272,7 @@
     <string name="text_clock">World Clock</string>
     <string name="select_timezone">Select Timezone</string>
     <string name="search_timezone">Search Timezone..</string>
-
+    <string name="time_zone">Asia/Kolkata</string>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: we already have this string in strings.xml as sample_timezone
why another string with same value?

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java (hunk 4 above)
Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/WorldClockActivity.java]: Dont use another library. use native code only to fix this issue

**Comment by HeinKhantZaw** [comment]
Shall I close this PR and then re submit again?

**Comment by Swati4star** [comment]
@HeinKhantZaw Whichever  way you are comfortable. Just take care of all the review comments mentioned by me.

**Comment by HeinKhantZaw** [comment]
Ok

**Pull request closed without merging**